File: backend/src/main/java/com/example/estimateserver/model/ContactMethod.java
package com.example.estimateserver.model;

import jakarta.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "contact_methods")
public class ContactMethod {
    @Id
    private String id = UUID.randomUUID().toString();

    @Column(nullable = false)
    private String contactId;

    @Column(nullable = false)
    private String methodType;

    @Column(nullable = false)
    private String value;

    @Column(nullable = false)
    private String userId;

    public ContactMethod() {}

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getContactId() { return contactId; }
    public void setContactId(String contactId) { this.contactId = contactId; }

    public String getMethodType() { return methodType; }
    public void setMethodType(String methodType) { this.methodType = methodType; }

    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }
}

File: app/src/main/java/com/example/myapplication/data/models/ContactMethod.kt
package com.example.myapplication.data.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(
    tableName = "contact_methods",
    foreignKeys = [
        ForeignKey(
            entity = Contact::class,
            parentColumns = ["id"],
            childColumns = ["contactId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["contactId"])]
)
data class ContactMethod(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),
    val contactId: String,
    val methodType: String,
    val value: String,
    val userId: String = ""
)